presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaFilterManager.java

```diff
@@ -0,0 +1,276 @@
+import static io.prestosql.plugin.kafka.KafkaInternalFieldDescription.OFFSET_TIMESTAMP_FIELD;
+import static io.prestosql.plugin.kafka.KafkaInternalFieldDescription.PARTITION_ID_FIELD;
+import static io.prestosql.plugin.kafka.KafkaInternalFieldDescription.PARTITION_OFFSET_FIELD;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static org.testng.Assert.assertEquals;
+
+public class TestKafkaFilterManager
+{
+    private static final int PARTITION_0 = 0;
+    private static final int PARTITION_1 = 1;
+    private static final int PARTITION_2 = 2;
+    private static final long START_TIMESTAMP = 1599035000;
+    private static final long END_TIMESTAMP = 1599035650;
+    private static final String TOPIC_NAME = "topic1";
+    private static final String SCHEMA_NAME = "kafka";
+
+    private final Map<Integer, Range> defaultPartitionValues = ImmutableMap.of(
+            PARTITION_0, new Range(10, 100),
+            PARTITION_1, new Range(11, 200),
+            PARTITION_2, new Range(12, 300));
+    private final Range offsetFilterPartitionValues = new Range(20, 80);
+    private final Map<Integer, Range> timestampFilterPartitionValues = ImmutableMap.of(
+            PARTITION_0, new Range(30, 70),
+            PARTITION_1, new Range(31, 170),
+            PARTITION_2, new Range(32, 270));
+
+    @Test
+    public void testGetKafkaFilter()
+    {
+        // test partition only
+        testFilter(true, false, false, false,
+                filter -> assertEquals(filter.getPartitionInfos().size(), defaultPartitionValues.size()),
+                filter -> {
+                    assertEquals(filter.getPartitionInfos().size(), 1);
+                    assertEquals(filter.getPartitionInfos().get(0).partition(), 1);
+                });
+
+        // test offset only
+        testFilter(false, true, false, false,
+                filter -> {
+                    assertEquals((long) filter.getPartitionBeginOffsets().get(new TopicPartition(TOPIC_NAME, PARTITION_0)), defaultPartitionValues.get(PARTITION_0).getBegin());
+                    assertEquals((long) filter.getPartitionEndOffsets().get(new TopicPartition(TOPIC_NAME, PARTITION_0)), defaultPartitionValues.get(PARTITION_0).getEnd());
+                    assertEquals((long) filter.getPartitionBeginOffsets().get(new TopicPartition(TOPIC_NAME, PARTITION_1)), defaultPartitionValues.get(PARTITION_1).getBegin());
+                    assertEquals((long) filter.getPartitionEndOffsets().get(new TopicPartition(TOPIC_NAME, PARTITION_1)), defaultPartitionValues.get(PARTITION_1).getEnd());
+                    assertEquals((long) filter.getPartitionBeginOffsets().get(new TopicPartition(TOPIC_NAME, PARTITION_2)), defaultPartitionValues.get(PARTITION_2).getBegin());
+                    assertEquals((long) filter.getPartitionEndOffsets().get(new TopicPartition(TOPIC_NAME, PARTITION_2)), defaultPartitionValues.get(PARTITION_2).getEnd());
```

Comment:
We're making this set of assertions multiple times.  Since the filter is the same every time (result of `buildDefaultKafkaFilter()`), it would seem sufficient to assert just once (if we need them at all).